chainer/functions/connection/linear.py

```diff
@@ -109,7 +132,13 @@ def forward(self, inputs):
                 1 in gy.shape):
             gy = numpy.ascontiguousarray(gy)
 
-        gW = gy.T.dot(x).astype(gy.dtype, copy=False)
+        if self._n_batch_axes > 1:
+            xp = cuda.get_array_module(*inputs)
+            ax = six.moves.range(self._n_batch_axes)
```

Comment:
Is it OK to use iterator here?

Reply:
Thank you for pointing out it. I checked the code below, and it works.

```
Python 3.6.4 |Anaconda, Inc.| (default, Dec 21 2017, 21:42:08)

In [2]: import numpy as np
In [3]: a = np.random.rand(1, 2, 3)
In [4]: b = np.random.rand(2, 3, 4)
In [6]: np.tensordot(a, b, axes=(range(1, 3), range(2)))
Out[6]: array([[1.77096642, 1.14276205, 1.64909544, 1.54107583]])
```

So I think it's OK to use iterator here, the documentation of numpy.tensordot just says "array_like" for the argument. Should I enclose this with `list()` or `tuple()` ?

Reply:
Ok, thank you. You are right.